c7n/resources/s3.py

```diff
@@ -2762,6 +2763,237 @@ def get_inventory_delta(self, client, inventory, b):
+                resource: s3
+                filters:
+                  - type: intelligent-tiering
+                    attrs:
+                      - Status: Enabled
+                      - Filter:
+                          And:
+                            Prefix: test
+                            Tags:
+                              - Key: Owner
+                                Value: c7n
+                      - Tierings:
+                          - Days: 100
+                          - AccessTier: ARCHIVE_ACCESS
+
+    """
+    schema = type_schema(
+        'intelligent-tiering',
+        attrs={'$ref': '#/definitions/filters_common/list_item_attrs'},
+        count={'type': 'number'},
+        count_op={'$ref': '#/definitions/filters_common/comparison_operators'}
+    )
+    permissions = ('s3:GetIntelligentTieringConfiguration',)
+    annotation_key = "c7n:IntelligentTiering"
+    annotate_items = True
+
+    def __init__(self, data, manager=None):
+        super().__init__(data, manager)
+        self.data['key'] = self.annotation_key
+
+    def process(self, buckets, event=None):
+        with self.executor_factory(max_workers=2) as w:
+            futures = {w.submit(self.get_item_values, b): b for b in buckets}
+            for future in as_completed(futures):
+                b = futures[future]
+                if future.exception():
+                    self.log.error("Message: %s Bucket: %s", future.exception(), b['Name'])
+                    continue
+        return super().process(buckets, event)
+
+    def get_item_values(self, b):
+        if self.annotation_key not in b:
+            client = bucket_client(local_session(self.manager.session_factory), b)
+            try:
+                int_tier_config = client.list_bucket_intelligent_tiering_configurations(
+                    Bucket=b['Name'])
+                b[self.annotation_key] = int_tier_config.get(
+                    'IntelligentTieringConfigurationList', [])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'AccessDenied':
+                    method = 'list_bucket_intelligent_tiering_configurations'
+                    log.warning(
+                        "Bucket:%s unable to invoke method:%s error:%s ",
+                          b['Name'], method, e.response['Error']['Message'])
+                    b.setdefault('c7n:DeniedMethods', []).append(method)
+        return b.get(self.annotation_key)
+
+
+@actions.register('set-intelligent-tiering')
+class ConfigureIntelligentTiering(BucketActionBase):
+    """Action applies an intelligent tiering configuration to a S3 bucket
+
+    The schema to supply to the configuration follows the schema here:
+     https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/put_bucket_intelligent_tiering_configuration.html
+
+    To delete a configuration, supply Status=delete with the either the Id or Id: matched
```

Comment:
I _think_ `Id: matched` will work OK, always curious if/how magic strings can cause problems though. Looks like this would only be confusing if there was somehow an intelligent tiering config named `matched` that you _didn't_ want to delete. Is that off the wall edge case worth using something like `matched: true` instead? ...probably not :sweat_smile:

Reply:
Thanks @ajkerrigan. Appreciate your feedback. Yes, that's something I was also thinking about 😄 . However, we use this pattern in various other actions like `remove-statements`, `set-permissions` on ebs volumes and I wanted to keep it consistent. Wanted to expose a way that could delete intelligent-tiering configurations as they change over time. Also, I tried to make it clear in the docs 😄 .